File: front-end/src/application/usecase/workflow/workflow-delete-use-case.ts
import { WorkFlow } from '@/domain/entity/workflow';
import { UllUUID } from '@ull-tfg/ull-tfg-typescript';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';

export interface WorkFlowDeleteUseCase {

    /**
     * Delete a workflow
     * 
     * @param workflowId - The UUID of the workflow to delete
     * @returns The deleted workflow, if found
     */
    deleteWorkFlow(workflowId: UllUUID): Promise<Either<DataError, WorkFlow>>;
}

File: back-end/src/main/java/es/ull/project/application/usecase/workflow/WorkFlowDeleteUseCase.java
package es.ull.project.application.usecase.workflow;

import java.util.Optional;
import java.util.UUID;

import es.ull.project.domain.entity.WorkFlow;

public interface WorkFlowDeleteUseCase {
    
    /**
     * Delete an workflow
     * 
     * @param workflow
     * @return workflow deleted
     */
    Optional<WorkFlow> deleteWorkFlow(UUID workflow);
}
